crates/marine-test-macro-impl/src/marine_test/glue_code_generator.rs

```diff
@@ -185,33 +191,89 @@ fn generate_test_glue_code_single_service(
 }
 
 fn generate_test_glue_code_multiple_services(
-    func_item: syn::ItemFn,
+    item: syn::Item,
     services: HashMap<String, ServiceDescription>,
     test_file_path: PathBuf,
 ) -> TResult<TokenStream> {
-    let service_definitions =
-        token_stream_generator::generate_service_definitions(services, &test_file_path)?;
+    let service_definitions = token_stream_generator::generate_service_definitions(
```

Comment:
Looks like it possible to change a signature of the `generate_service_definitions` function to recieve iterator instread of entire `HashMap`.